flux-kube/plugins/flux-kube.c

```diff
@@ -0,0 +1,352 @@
+                                             "flux-kube::jobspec")))
+        return 0;
+
+    if (!(f = flux_rpc_pack (flux_jobtap_get_flux (p), "fluxkube.cancel", 
+                             FLUX_NODEID_ANY, 0,
+                             "{s:s}", "jobspec", jobspec))) { 
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to send RPC: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+    
+    if (flux_future_then (f, -1.0f, get_response_continuation, p) < 0) {
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to "
+                                     "cancel jobspec: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+
+    /*  If flux-kube::free is set on this job, then this plugin
+     *  sent an "alloc" event, so a "free" event needs to be sent now.
+     */
+    if (flux_jobtap_job_aux_get (p,
+                                 FLUX_JOBTAP_CURRENT_JOB,
+                                 "flux-kube::free")) {
+        if (flux_jobtap_event_post_pack (p,
+                                         FLUX_JOBTAP_CURRENT_JOB,
+                                         "free",
+                                         NULL) < 0)
+             flux_log_error (flux_jobtap_get_flux (p),
+                             "flux-kube: failed to post free event");
+    }
+    return 0;
+}
+
+static int validate_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *arg)
+{
+    int fluxkube = 0, saved_errno = 0;
+    json_t *jobspec = NULL, *R = NULL;
+    char *js = NULL, *Rstring = NULL;
+    uint32_t userid = UINT32_MAX;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i s:{s:{s:{s?i}}}}",
+                                "userid", &userid,
+                                "jobspec",
+                                 "attributes",
+                                  "system",
+                                   "flux-kube", &fluxkube) < 0) {
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "invalid system.flux-kube: %s",
+                                       flux_plugin_arg_strerror (args));
+    }
+
+    //  Nothing to do if not fluxkube
+    if (!fluxkube)
+        return 0;
+
+    if (userid != getuid ())
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "Guest user cannot use alloc bypass");
```

Comment:
Suggestion: Change error message to `"Guest user cannot set flux-kube attribute"`